staging/src/k8s.io/apiserver/pkg/server/filters/timeout.go

```diff
@@ -119,15 +120,14 @@ func (t *timeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			// the work needs to send to it. This is defer'd to ensure it runs
 			// ever if the post timeout work itself panics.
 			go func() {
+				timedOutAt := time.Now()
 				res := <-resultCh
```

Comment:
why does this block forever.  I would expect to see similar: "this gofunc must end" logic.

Reply:
this is in keeping with the current implementation of the timeout filter. If I make this change I would like to add a test for it, and a proper test necessitates a refactor of the timeout filter. 
@deads2k I would like to do a follow up PR for this. Let me know if you want me to address it in this PR.